Automattic/jetpack
# Add initial state to the connection package

This PR proposes a way for the Connection composer package to expose an Initial State that can be consumed by the connection JS package. This will make it easier for consumers to use the connection js package and not have to worry about its initial state.

For now this is working for the ConnectionStatus, but it can also work for the `apiRoot` and `registrationNonce`.

#### Changes proposed in this Pull Request:

* Add initial state to the connection package

#### Jetpack product discussion


N/A

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






This is a bit tricky to test because, currently, the `getConnectionStatus` resolver will always fetch the connection status from the API. Unless we merge https://github.com/Automattic/jetpack/pull/21721.

So, first thing:

* Heads up to https://github.com/Automattic/jetpack/pull/21721 and copy the modifications to your environment. or cherry pick that commit.
* Build Jetpack and Backup plugins
* Disconnect
* Visit both plugins pages and make sure the request to `v4/connection` endpoint is NOT performed.

Discussion:

We can either merge https://github.com/Automattic/jetpack/pull/21721 or a similar approach, or we can enforce that the connection status MUST be informed via the initial state and make all consumers use this approach. Then we could safely remove the resolver.

What do you think?
